### PR TITLE
[1LP][RFR]Automated test_create_sns_topic

### PR DESCRIPTION
**Pull request by mmojzis**
{{ pytest: -k "test_create_sns_topic" -v }}

### CODE REVIEW

File: cfme/tests/cloud/test_providers.py [hunk 1]
```diff
@@ -1238,8 +1238,8 @@ def test_ec2_api_filter_limit():
 
 
 @test_requirements.ec2
-@pytest.mark.manual
-def test_ec2_create_sns_topic():
+@pytest.mark.provider([EC2Provider], scope="function", override=True, selector=ONE)
```
Comment (john-dupuy) [inline cfme/tests/cloud/test_providers.py]: `override=True` not necessary here.

**Comment by mmojzis** [comment]
> Maybe I am missing something but I don't see what this is testing in CFME. It looks like everything is done via the EC2 api.
> 
> Does `creating`/adding the provider to CFME create new_topic? Is that what we're testing here? If so, can you add a comment or modify the test steps to reflect this.

Added comments there so it should be more clear what is this test testing.

File: cfme/tests/cloud/test_providers.py (hunk 1 above)
Comment (mmojzis) [reply cfme/tests/cloud/test_providers.py]: Deleted that.

File: cfme/tests/cloud/test_providers.py [hunk 2]
```diff
@@ -1256,7 +1256,19 @@ def test_ec2_create_sns_topic():
             1.
             2. Check SNS topic for this region in AWS
     """
-    pass
+    # preparations for test
+    request.addfinalizer(lambda: provider.delete_if_exists())
```
Comment (jaryn) [inline cfme/tests/cloud/test_providers.py]: Jut note: This is enough to do the same job.
```suggestion
    request.addfinalizer(provider.delete_if_exists)
```

File: cfme/tests/cloud/test_providers.py [hunk 3]
```diff
@@ -1256,7 +1256,19 @@ def test_ec2_create_sns_topic():
...
+    topic = provider.mgmt.get_arn_if_topic_exists('AWSConfig_topic')
+    if topic:
+        provider.mgmt.delete_topic(topic)
+
+    # SNS topic should be automatically created during provider creation in CFME
+    provider.create()
+    new_topic = wait_for(lambda: provider.mgmt.get_arn_if_topic_exists('AWSConfig_topic'), delay=15,
```
Comment (jaryn) [inline cfme/tests/cloud/test_providers.py]: Note you don't need the lambda here as well as you can tell `wait_for` what arguments to use but yeah who does remember all the arguments for the `wait_for` .. (:

File: cfme/tests/cloud/test_providers.py [hunk 4]
```diff
@@ -1256,7 +1256,19 @@ def test_ec2_create_sns_topic():
             1.
             2. Check SNS topic for this region in AWS
     """
-    pass
+    # preparations for test
+    request.addfinalizer(lambda: provider.delete_if_exists())
+    topic = provider.mgmt.get_arn_if_topic_exists('AWSConfig_topic')
```
Comment (jaryn) [inline cfme/tests/cloud/test_providers.py]: I am confused
```
             2. Check SNS topic for this region in AWS
```
If the point of this test is checking the topic should exists, you need to assert it is present and not just delete it immediately.

If you need clean environment before the test does the actual testing, then I think either you need some janitor job somewhere which would clean-up old stuff or I think less-preferably you need to put the code deleting the unwanted elements in the fixture.

**Comment by john-dupuy** [comment]
> > Maybe I am missing something but I don't see what this is testing in CFME. It looks like everything is done via the EC2 api.
> > Does `creating`/adding the provider to CFME create new_topic? Is that what we're testing here? If so, can you add a comment or modify the test steps to reflect this.
> 
> Added comments there so it should be more clear what is this test testing.

Ah thanks, that answers my question.

File: cfme/tests/cloud/test_providers.py [hunk 5]
```diff
@@ -1256,7 +1256,19 @@ def test_ec2_create_sns_topic():
             1.
             2. Check SNS topic for this region in AWS
```
Comment (jaryn) [inline cfme/tests/cloud/test_providers.py]: It would be better if this read "Check SNS topic **exists** for this region in AWS". I got confused

Comment (mmojzis) [reply cfme/tests/cloud/test_providers.py]: You are right! Added exists to expectedResults.

File: cfme/tests/cloud/test_providers.py (hunk 2 above)
Comment (mmojzis) [reply cfme/tests/cloud/test_providers.py]: Changed it.

File: cfme/tests/cloud/test_providers.py (hunk 3 above)
Comment (mshriver) [reply cfme/tests/cloud/test_providers.py]: We have a bunch of places where this is true, I've tried to push towards using func_args more :)

**Pull request merged**